src/client.rs

```diff
@@ -1280,6 +1280,48 @@ impl<T: Read + Write> Session<T> {
         .and_then(|lines| parse_id_seq(&lines, &mut self.unsolicited_responses_tx))
     }
 
+    /// The [`SETQUOTA` command](https://datatracker.ietf.org/doc/html/rfc2087#section-4.1)
+    ///
+    /// Updates the resource limits for a mailbox. Any previous limits for the named quota
+    /// are discarded.
+    /// Returns the updated quota
+    pub fn set_quota<S: AsRef<str>>(
+        &mut self,
+        quota_root: S,
+        limits: Vec<QuotaLimit>,
+    ) -> Result<Quota> {
+        self.run_command_and_read_response(&format!(
+            "SETQUOTA {} ({})",
+            validate_str("SETQUOTA", "quota_root", quota_root.as_ref())?,
+            limits
+                .iter()
+                .map(|e| e.to_string())
```

Comment:
This, too, feels unfortunate. Technically we don't need `String`s, all we need is to `write!("{}", e)` into the string we already have. Which you can do with `write!(&mut somestring, "{}", e)` as long as `std::fmt::Write` is in scope.

Reply:
I switched it to the cargo iter_into (as well as the other usage I found)